data_steward/cdr_cleaner/manual_cleaning_rules/dc732_suppress.py

```diff
@@ -0,0 +1,445 @@
+            backup_table_query_job = CLIENT.query(backup_table_rows_query, job_config=job_config)
+            query_jobs.append(backup_table_query_job)
+            
+            if ext_table_name:
+                backup_ext_dest_table = f'{PROJECT_ID}.{SANDBOX_DATASET_ID}.{ISSUE_PREFIX}{ext_table_name}'
+                job_config = bigquery.QueryJobConfig(destination=backup_ext_dest_table)
+                job_config.write_disposition = 'WRITE_APPEND'
+                backup_ext_table_rows_query = BACKUP_EXT_TABLE_ROWS_QUERY.render(dataset_ids=dataset_ids, table=table_name, suppress_rows_table=suppress_rows_table)
+                backup_ext_table_query_job = CLIENT.query(backup_ext_table_rows_query, job_config=job_config)
+                query_jobs.append(backup_ext_table_query_job)
+    return query_jobs
+
+
+# -
+
+# # Execute Deletion
+# Use a meta-query to generate the minimal set of DELETE statements needed to delete rows in domain and \_ext tables.
+
+# +
+GET_DELETE_QUERIES = JINJA_ENV.from_string("""
+WITH table_ids AS
+(SELECT DISTINCT dataset_id, table 
+FROM `{{suppress_rows_table}}`)
+SELECT CONCAT('DELETE FROM `', 
+ dataset_id, '.', table, '` t', 
+ ' WHERE EXISTS', 
+ '(SELECT 1 FROM {{suppress_rows_table}} WHERE dataset_id = "', dataset_id, '" AND table = "', table, '" AND t.', table, '_id = row_id)')
+AS query_string
+FROM table_ids
+""")
+
+def get_delete_queries(suppress_rows_table):
+    """
+    Get list of DELETE statements needed to delete rows in domain and _ext tables
+    
+    :param suppress_rows_table: identifies the table where suppress rows are loaded
+    """
+    get_delete_query = GET_DELETE_QUERIES.render(suppress_rows_table=suppress_rows_table)
+    delete_query_job = CLIENT.query(get_delete_query)
+    delete_query_results = list(delete_query_job.result())
+    all_delete_queries = []
+    for delete_query_result in delete_query_results:
+        delete_query = delete_query_result['query_string']
+        # HACK to get associated _ext DELETE statement
+        delete_ext_query = delete_query.replace('` t WHERE', '_ext` t WHERE')
+        all_delete_queries.extend([delete_query, delete_ext_query])
+    return all_delete_queries
+
+
+# -
+
+def run_delete_queries(delete_queries):
+    """
+    Execute list of queries which delete rows
+    
+    :param delete_queries: list of DELETE statements
+    """
+    # Delete rows in batch to prevent concurrent request limit
+    delete_query_jobs = []
+    for delete_query in delete_queries:
+        print(delete_query + '\n')
+        job_config = bigquery.QueryJobConfig(priority=bigquery.QueryPriority.BATCH)
+        delete_query_job = CLIENT.query(delete_query, job_config=job_config)
+        delete_query_jobs.append(delete_query_job)
+    return delete_query_jobs
+
+
+def print_jobs(delete_query_jobs):
+    # Manually checking if jobs are all done...
```

Comment:
you could generalize this function by changing the parameter name.  I think this information would be relevant to other types of BigQuery jobs also.